include/nbl/builtin/hlsl/cpp_compat/cpp_compat.hlsl

```diff
@@ -17,6 +20,15 @@ using add_reference = std::add_lvalue_reference<T>;
 template<typename T>
 using add_pointer = std::add_pointer<T>;
 
+// TODO[Przemog]: is there smarter way to do that? 
+// TODO[Przemog]: move functions below to other file (in cpp_compat drectory?)
+template<typename T, typename U>
+inline T lerp(const T& lhs, const T& rhs, const U& t)
+{
+    return glm::mix(lhs, rhs, t);
+}
```

Comment:
`lerp` is a HLSL intrinsic, so lets move all the intrinsics implementations for C++ to `cpp_compat/intrinsics.h`

P.S. also make sure to explicitly specify the template parameters when calling `glm::` functions

Reply:
ok, nvm I see that `glm::` functions have vastly different template parameters

Reply:
btw I made it myself on `master`